### PR TITLE
Do not export GOROOT for Go versions >= 1.9

### PR DESCRIPTION
**Pull request by mmlb**
**Description:**

Do not export GOROOT. This has not been necessary since [Go 1.9](https://go.dev/doc/go1.9#goroot) at least (although clunky) but definitely not since [Go 1.10](https://go.dev/doc/go1.10#goroot). This is cargo culting code that is more than 2 years out of date and runs into issues when multiple go versions are used in an action run.

**Related issue:**

#107 

**Check list:**
- [ ] Mark if documentation changes are required.
- [x] Mark if tests were added or updated to cover the changes.

### CODE REVIEW

**Comment by mmlb** [comment]
Opened this up as a draft to get CI running as I'm a super n00b at ts/js.

**Comment by mmlb** [comment]
The actual fix commit I'll add afterwards is:
```
commit 607cfc72940f41ed709d01f36a75387ed7d150dd
Author: Manuel Mendez <mmendez534@gmail.com>
Date:   Tue Dec 14 13:24:18 2021 -0500

    Do not export GOROOT
    
    This has not been necessary since [Go 1.9](https://go.dev/doc/go1.9#goroot) at least (although clunky) but definitely not since [Go 1.10](https://go.dev/doc/go1.10#goroot).
    This is cargo culting code that is more than 2 years out of date and runs into issues when multiple go versions are used in an action run.
    
    Signed-off-by: Manuel Mendez <mmendez534@gmail.com>

diff --git a/__tests__/setup-go.test.ts b/__tests__/setup-go.test.ts
index a001839..9368804 100644
--- a/__tests__/setup-go.test.ts
+++ b/__tests__/setup-go.test.ts
@@ -243,7 +243,7 @@ describe('setup-go', () => {
     });
 
     await main.run();
-    expect(vars).toBe({'GOROOT': 'foo'});
+    expect(vars).toBe({});
   });
 
   it('finds a version of go already in the cache', async () => {
diff --git a/src/main.ts b/src/main.ts
index 2d90b2f..29ae2b6 100644
--- a/src/main.ts
+++ b/src/main.ts
@@ -26,7 +26,6 @@ export async function run() {
 
       const installDir = await installer.getGo(versionSpec, stable, auth);
 
-      core.exportVariable('GOROOT', installDir);
       core.addPath(path.join(installDir, 'bin'));
       core.info('Added go to the path');
```

**Comment by mmlb** [comment]
Pinging @bryanmacfarlane  and @joshmgross because it seems the `spark` I pinged isn't the same `spark` as in reviewers.

**Comment by mmlb** [comment]
Hi @joshmgross I've updated to address a merge conflict. It'd be really nice if someone could take a look here.

**Comment by joshmgross** [comment]
👋  @mmlb could you take a look at the failing CI?

It looks like we need to update `dist.js`with `npm run build` and there's a formatting issue

**Comment by mmlb** [comment]
Hey @joshmgross yep will fix. I remember now that I expected ci to fail for this reason, I couldn't figure out how to get tests running locally.

**Comment by mmlb** [comment]
@joshmgross updated!

File: __tests__/setup-go.test.ts [hunk 1]
```diff
@@ -230,6 +231,22 @@ describe('setup-go', () => {
...
+    let toolPath = path.normalize('/cache/go/1.13.0/x64');
+    findSpy.mockImplementation(() => toolPath);
+
+    let vars = {} as any;
+    exSpy.mockImplementation(async (name, val) => {
+      vars[name] = val;
+    });
+
+    await main.run();
+    expect(vars).toBe({});
```
Comment (joshmgross) [inline __tests__/setup-go.test.ts:247]: Looks like the test is failing

```
    expect(received).toBe(expected) // Object.is equality

    If it should pass with deep equality, replace "toBe" with "toStrictEqual"
```suggestion
    expect(vars).toBe({});
```
```suggestion
    expect(vars).toStrictEqual({});
```

Does this fix it?

Comment (joshmgross) [inline __tests__/setup-go.test.ts:247]: Looks like the test is failing

```
    expect(received).toBe(expected) // Object.is equality

    If it should pass with deep equality, replace "toBe" with "toStrictEqual"
```suggestion
    expect(vars).toBe({});
```
```suggestion
    expect(vars).toStrictEqual({});
```

Does this fix it?

File: __tests__/setup-go.test.ts [hunk 2]
```diff
@@ -230,6 +231,22 @@ describe('setup-go', () => {
     expect(logSpy).toHaveBeenCalledWith(`Setup go version spec 1.13.0`);
   });
 
+  it('exports GOROOT', async () => {
```
Comment (brcrista) [inline __tests__/setup-go.test.ts:234]:
```suggestion
  it('does not export any variables', async () => {
```

Looks like the test is making sure we *don't* export `GOROOT`, correct?

File: __tests__/setup-go.test.ts (hunk 1 above)
Comment (mmlb) [reply __tests__/setup-go.test.ts]: Yep that fixed it, I've updated the PR.

**Comment by mmlb** [comment]
Thanks for the review to both of you. I'm not a ts/js dev so had no clue how to run/test this and was letting CI do, but using the `npm run build` as a hint I figured out `npm run test` :D.

**Comment by joshmgross** [comment]
It looks like CI is failing for validating `setup-go` with versions < 1.9 that still need `GOROOT` - https://github.com/actions/setup-go/runs/5500146663?check_suite_focus=true

It seems like we should still export GOROOT if go-version is < 1.9.
@mmlb can you made those changes? If you're not comfortable enough with JS/TS, I'd be happy to help out.

File: __tests__/setup-go.test.ts [hunk 3]
```diff
@@ -230,6 +231,22 @@ describe('setup-go', () => {
     expect(logSpy).toHaveBeenCalledWith(`Setup go version spec 1.13.0`);
   });
 
+  it('doest not export any varibles', async () => {
```
Comment (brcrista) [inline __tests__/setup-go.test.ts:234]:
```suggestion
  it('does not export any variables', async () => {
```

Comment (mmlb) [reply __tests__/setup-go.test.ts]: fixed

**Comment by mmlb** [comment]
> It looks like CI is failing for validating `setup-go` with versions < 1.9 that still need `GOROOT` - [actions/setup-go/runs/5500146663?check_suite_focus=true](https://github.com/actions/setup-go/runs/5500146663?check_suite_focus=true)
> 
> It seems like we should still export GOROOT if go-version is < 1.9. @mmlb can you made those changes? If you're not comfortable enough with JS/TS, I'd be happy to help out.

I'd appreciate if you took over on that change.

Side question: go1.8 is > 5years old, whats the policy for setup-go dropping versions? Is it "don't, unless *really* *really* necessary"?

**Comment by joshmgross** [comment]
>Side question: go1.8 is > 5years old, whats the policy for setup-go dropping versions? Is it "don't, unless really really necessary"?

I don't know if we have an explicit policy at this point - we'd need more data to understand what versions users need in their workflows.

File: __tests__/setup-go.test.ts [hunk 4]
```diff
@@ -230,6 +231,22 @@ describe('setup-go', () => {
     expect(logSpy).toHaveBeenCalledWith(`Setup go version spec 1.13.0`);
   });
 
+  it('does not export any varibles', async () => {
```
Comment (brcrista) [inline __tests__/setup-go.test.ts:234]:
```suggestion
  it('does not export any variables', async () => {
```

Comment (brcrista) [reply __tests__/setup-go.test.ts:234]: FYI you can just commit suggestions directly from the GitHub UI:

Screen Shot 2022-03-11 at 2 10 50 PM

Comment (joshmgross) [reply __tests__/setup-go.test.ts]: Fixed with the changes in https://github.com/actions/setup-go/pull/175/commits/2a34c33bd7648b12fa2caabb0bfc8919e37aacc6

File: __tests__/setup-go.test.ts [hunk 5]
```diff
@@ -49,6 +50,8 @@ describe('setup-go', () => {
     inSpy.mockImplementation(name => inputs[name]);
     getBooleanInputSpy = jest.spyOn(core, 'getBooleanInput');
     getBooleanInputSpy.mockImplementation(name => inputs[name]);
+    exportVarSpy = jest.spyOn(core, 'exportVariable');
+    extractTarSpy = jest.spyOn(core, 'exportVariable');
```
Comment (mmlb) [inline __tests__/setup-go.test.ts:54]: is this a copy/paste error? Both are `exportVariable`, extractTarSpy should be `extractTar`.

**Review by mmlb — COMMENTED** [review]
thanks for taking this up, just one small nit.

**Review by brcrista — APPROVED** [review]

**Pull request merged**